framework/src/postprocessors/GeneralSensorPostprocessor.C

```diff
@@ -0,0 +1,315 @@
+  params.addParam<FunctionName>("delay_function", 0.0,
+                                "Delay function describing the delay over time");
+  params.addParam<FunctionName>("uncertainty_std_dev_function", 0.1,
+                                "Standard deviation of uncertainty function describing uncertainty std dev over time");
+  params.addParam<Real>("proportional_weight", 0.5, "The weight assigned to the proportional term");
+  params.addParam<Real>("integral_weight", 0.5, "The weight assigned to the integral term");
+  return params;
+}
+
+GeneralSensorPostprocessor::GeneralSensorPostprocessor(const InputParameters & parameters)
+  : GeneralPostprocessor(parameters),
+    _input_signal(getPostprocessorValue("input_signal")),
+    _drift_function(getFunction("drift_function")),
+    _scaling_factor(getParam<Real>("scaling_factor")),
+    _vector_size(getParam<Real>("vector_size")),
+    _pp_old(getPostprocessorValueOld("input_signal")),
+    _efficiency_function(getFunction("efficiency_function")),
+    _noise_std_dev_function(getFunction("noise_std_dev_function")),
+    _delay_function(getFunction("delay_function")),
+    _signalToNoise_function(getFunction("signalToNoise_function")),
+    _uncertainty_std_dev_function(getFunction("uncertainty_std_dev_function")),
+    _proportional_weight(getParam<Real>("proportional_weight")),
+    _integral_weight(getParam<Real>("integral_weight"))
+{
+}
+
+void GeneralSensorPostprocessor::initialize() 
+{ 
+  _time_values.push_back(_t);
+  _input_signal_values.push_back(_input_signal);
+
+  // Check if the size is greater than 500
+  if (_time_values.size() > 500 && _input_signal_values.size() > _vector_size) 
+  {
+    // Remove the first 10 elements
+    _time_values.erase(_time_values.begin(), _time_values.begin() + 10);
+    _input_signal_values.erase(_input_signal_values.begin(), _input_signal_values.begin() + 10);
+  }
+
+
+
+  //-------------DRIFT-------------
+  // Evaluate drift function at the current time
+  Real drift_value = 0.0;
+  drift_value = _drift_function.value(_t);
+
+  //-------------EFFICIENCY-------------
+  // Efficiency function at the current time
+  Real efficiency_value = 0.8;    // default efficiency is taken as 80%
+  efficiency_value = _efficiency_function.value(_t); 
+
+  //-------------SIGNAL TO NOISE FACTOR-------------
+  // Signal to noise factor as a function of time
+  Real signalToNoise_value = 0.2; // default value in MOOSE
+  signalToNoise_value = _signalToNoise_function.value(_t);
+
+  //-------------NOISE-------------
+  // Sample a normal distribution centered around the noise mean, with a width of the noise standard deviation
+  Real noise_value = 0.0;
+  // noise std dev as a function of time
+  Real noise_std_dev = 0.5; // default value
+  noise_std_dev = _noise_std_dev_function.value(_t);
+  //Real random_number = Sampler::getRand1(1,0,1); // NO NEED
+  //MooseRandom::seed(10); 

```

Comment:
in the .h file add
```
MooseRandom _rng;
```

and in the constructor do

```
_rng.seed(0, getParam<unsigned int>("seed");
```
(and add a "seed" input parameter)

down here pull a random number using

```
_rng.randNormal(0, 0, noise_std_dev);
```